backend/variants/models/scores.py

```diff
@@ -546,6 +657,81 @@ def unroll_extra_annos_result(rows, fields):
     return rows_
 
 
+def generate_pedia_input(self, pathoEnabled, prioEnabled, gmEnabled, queryId, case_name, results):
+    pathogenicity_scores = None
+    if pathoEnabled:
+        pathogenicity_scores = {
+            (row.chromosome, row.start, row.reference, row.alternative): row.score
+            for row in SmallVariantQueryVariantScores.objects.filter(query__sodar_uuid=queryId)
+        }
+    phenotype_scores = None
+    if prioEnabled:
+        phenotype_scores = {
+            row.gene_id: row.score
+            for row in SmallVariantQueryGeneScores.objects.filter(query__sodar_uuid=queryId)
+            if row.gene_id
+        }
+    gm_scores = None
+    if gmEnabled:
+        gm_scores = {
+            row.gene_id: 0 if row.score == 0 else row.score
+            for row in SmallVariantQueryGestaltMatcherScores.objects.filter(
+                query__sodar_uuid=queryId
+            )
+            if row.gene_id
+        }
+
+    payloadList = []
+    """Read and json object by reading ``results`` ."""
+    for line in results:
+        payload = dict()
+
+        if line["entrez_id"]:
+            if line["symbol"]:
+                payload["gene_name"] = line["symbol"]
+            else:
+                payload["gene_name"] = " "
+            payload["gene_id"] = line["entrez_id"]
+
+        if phenotype_scores and line.entrez_id:
+            payload["cada_score"] = phenotype_scores.get(line.entrez_id, -1)
+
+        if pathogenicity_scores:
+            payload["cadd_score"] = pathogenicity_scores.get(
+                (line.chromosome, line.start, line.reference, line.alternative), 0.0
+            )
+
+        if gm_scores and line.entrez_id:
+            payload["gestalt_score"] = (
+                0
+                if gm_scores.get(line.entrez_id, 0) == float("nan")
+                else gm_scores.get(line.entrez_id, 0)
+            )
+
+        payload["label"] = False
+
+        payloadList.append(payload)
+
+    df = pd.DataFrame(payloadList)
+
+    if "cadd_score" in df:
+        # Sort the DataFrame based on the 'cadd_score' column in descending order
+        df_sorted = df.sort_values(by="cadd_score", ascending=False)
+
+        # Drop duplicates in the 'gene_name' column, keeping the first occurrence (highest CADD score)
+        df_no_duplicates = df_sorted.drop_duplicates(subset="gene_name", keep="first")
+
+        if case_name.startswith("F_"):
```

Comment:
@ahujameg I don't understand this if condition. Is it specific to the Bonn case naming scheme?

Reply:
Yes. Before adding the if condition, it was not working for those cases whose case name starts with F_